src/CsvHelper/CsvParser.cs

```diff
@@ -356,34 +494,41 @@ private ReadLineResult ReadLine(ref char c, ref char cPrev)
 				}
 
 				var isFirstCharOfRow = rowStartPosition == bufferPosition - 1;
-				if (isFirstCharOfRow && (allowComments && c == comment || ignoreBlankLines && ((c == '\r' || c == '\n') && !isNewLineSet || c == newLineFirstChar && isNewLineSet)))
+
+				if (isFirstCharOfRow &&

```

Comment:
Too many conditions for one if statement, hard to understand, maybe we can create several bool variables, so we get something like this - 

```
bool4 = bool7 || bool8;
bool3 = (bool2 && bool3) || (bool4 && bool5);
if (isFirstCharOfRow && bool3)
{}
```